File: app/src/main/java/com/example/muestreo_fauna_salvaje/fragments/MenuPrincipal.java
package com.example.muestreo_fauna_salvaje.fragments;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.Navigation;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.example.muestreo_fauna_salvaje.R;
import com.example.muestreo_fauna_salvaje.viewModels.LocationViewModel;

public class MenuPrincipal extends Fragment {



    public MenuPrincipal() {
        // Required empty public constructor
    }

    public static MenuPrincipal newInstance() {
        MenuPrincipal fragment = new MenuPrincipal();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        iniciarServicioLocalizacion();
        return  inflater.inflate(R.layout.fragment_menu_principal, container, false);
    }

    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        Button btnAñadir = (Button)view.findViewById(R.id.btnAñadir);
        btnAñadir.setOnClickListener( new View.OnClickListener(){

            @Override
            public void onClick(View view) {
                Navigation.findNavController(view).navigate(R.id.action_menuPrincipal_to_avistamiento);
            }
        });
        ((Button) view.findViewById(R.id.btnResumen)).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Navigation.findNavController(view).navigate(R.id.action_menuPrincipal_to_resumen);
            }
        });
    }

    private void iniciarServicioLocalizacion() {
        LocationViewModel vwLocation = new ViewModelProvider(this).get(LocationViewModel.class);
        vwLocation.inicializar(getContext(),getActivity());
    }

}

File: Kotlin/MyApplication/app/src/main/java/com/example/myapplication/fragments/MenuPrincipal.kt
package com.example.myapplication.fragments

import android.annotation.SuppressLint
import android.os.Bundle
import android.util.Log
import android.view.*
import android.widget.*
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.viewpager2.widget.ViewPager2
import com.example.myapplication.R
import com.example.myapplication.data_classes.AvistamientoData
import com.example.myapplication.database.DatabaseInstance
import com.example.myapplication.viewmodels.DatabaseViewModel
import com.google.android.material.bottomappbar.BottomAppBar
import com.jakewharton.rxbinding2.view.clicks
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import kotlinx.android.synthetic.main.fragment_avistamiento.*
import java.util.*

class MenuPrincipal : Fragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_menu_principal, container, false)
    }


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val vmDatabase = ViewModelProviders.of(this).get(DatabaseViewModel(DatabaseInstance.getInstance(view.context))::class.java);
        vmDatabase.retrieveSightsings();

        vmDatabase.sightings.observe(viewLifecycleOwner, Observer {
                if(!it.isEmpty()) generateTable(view, it)
        })
    }

    private fun generateTable(view:View, animals: List<AvistamientoData>?) {
        val table = view.findViewById<TableLayout>(R.id.tableSeen);
        fun createTextView(text:String):TextView{
            var newTextView = TextView(context);
            newTextView.text = text;
            return  newTextView;
        }
        animals?.forEach({
            val newRow = TableRow(context);
            newRow.addView(createTextView(it.especie));
            newRow.addView(createTextView(it.latitude.toString()));
            newRow.addView(createTextView(it.longitude.toString()));
            newRow.addView(createTextView(it.date));
            newRow.addView(createTextView(it.time));
            table.addView(newRow)
        })


    }


}